modules/related-posts/jetpack-related-posts.php

```diff
@@ -175,12 +175,33 @@ public function filter_add_target_to_dom( $content ) {
 		}
 
 		if ( ! $this->_found_shortcode ) {
-			$content .= "\n" . $this->get_target_html();
+			if ( Jetpack_AMP_Support::is_amp_request() ) {
+				$content .= "\n" . $this->get_amp_html();
```

Comment:
Minor point, but there's nothing AMP-specific about this method. Probably better named `get_server_rendered_html()` with the other method called `get_client_rendered_html()`.

BTW, why not just always do server-side rendering?